builds/jenkins-build-sls-app/Jenkinsfile

```diff
@@ -0,0 +1,1062 @@
+        ]])
+    }
+
+    def configObj = dir(repo_name) {
+      LoadConfiguration()
+    }
+
+    if (configObj.service_id) {
+      config = serviceMetadataLoader.loadServiceMetadata(configObj.service_id)
+    } else {
+      error "Service Id is not available."
+    }
+  }
+
+  if (!config) {
+    error "Failed to fetch service metadata from catalog"
+  }
+
+  dir(repo_name) {
+    scmModule.setServiceConfig(config)
+    def envApi = "${g_base_url}/jazz/environments"
+    environmentDeploymentMetadata.initialize(config, configLoader, scmModule, branch, env.BUILD_URL, env.BUILD_ID, envApi, auth_token)
+    gitCommitHash = scmModule.getRepoCommitHash()
+    gitCommitOwner = scmModule.getRepoCommitterInfo(gitCommitHash)
+    context_map = [created_by: config['created_by'], deployed_by: gitCommitOwner]
+    environment_logical_id = '';
+
+    if (branch == 'master') {
+      current_environment = 'prod'
+      environment_logical_id = 'prod';
+    } else {
+      current_environment = 'dev'
+      environment_logical_id = environmentDeploymentMetadata.getEnvironmentLogicalId();
+    }
+
+    if (!environment_logical_id && config['domain'] != 'jazz') {
+      error "The environment has not been created yet and its missing the environment id"
+    }
+  }
+  if (!events) {
+    error "Can't load events module"
+  } //Fail here
+  def eventsApi = "${g_base_url}/jazz/events"
+  events.initialize(configLoader, config, "SERVICE_DEPLOYMENT", branch, environment_logical_id, eventsApi)
+
+  def runtime = config['providerRuntime']
+  def service = config['service']
+  def isScheduleEnabled = isEnabled(config, "eventScheduleRate")
+  def isStreamEnabled = isEnabled(config, "event_source_kinesis")
+  def isDynamoDbEnabled = isEnabled(config, "event_source_dynamodb")
+  def isS3EventEnabled = isEnabled(config, "event_source_s3")
+  def isSQSEventEnabled = isEnabled(config, "event_source_sqs")
+  def isEc2EventEnabled = isEnabled(config, "event_source_ec2")
+  def isEventSchdld = false
+  def internalAccess = config['require_internal_access']
+  domain = config['domain']
+
+  def roleId = config['iamRoleARN'].substring(config['iamRoleARN'].indexOf("::") + 2, config['iamRoleARN'].lastIndexOf(":"))
+
+  sonarModule.initialize(configLoader, config, branch)
+  stackName = "${configLoader.INSTANCE_PREFIX}-${config['domain']}-${config['service']}-${environment_logical_id}"
+
+  if (isScheduleEnabled || isEc2EventEnabled || isS3EventEnabled || isStreamEnabled || isDynamoDbEnabled || isSQSEventEnabled){
+    isEventSchdld = true
+  }
+
+  def requestId = utilModule.generateRequestId()
+  if (requestId != null) {
+    events.setRequestId(requestId)
+    environmentDeploymentMetadata.setRequestId(requestId)
+  } else {
+    error "Request Id Generation failed"
+  }
+
+  events.sendStartedEvent('CREATE_DEPLOYMENT', 'Function deployment started', environmentDeploymentMetadata.generateDeploymentMap("started", environment_logical_id, gitCommitHash), environment_logical_id)
```

Comment:
Update description of the event.

Reply:
Updated

Reply:
Resolved with https://github.com/tmobile/jazz/pull/829/commits/1ecbce6861cc64cb7698311ca4d3cef2070743f7